File: jscomp/test/demo_pipe.js
'use strict';


function register(rl) {
  return rl.on("line", (function (x) {
                  console.log(x);
                  
                })).on("close", (function (param) {
                console.log("finished");
                
              }));
}

exports.register = register;
/* No side effect */

File: jscomp/test/demo_pipe.ml
type readline
external on : readline ->
  ([ `line of string -> unit 
   | `close of unit -> unit] 
    [@string] ) ->
  readline = 
  "on" [@@send]
let register rl =
  rl
  |. on (`line (fun  x -> Js.log x ))
  |. on (`close (fun () -> Js.log "finished"))    

